### PR TITLE
Server to server

### PR DESCRIPTION
**Pull request by Kieaer**
server to server
Usage
```
String serverip = "localhost";
String port = 6568;
Call.onConnect(player.con, serverip, port);
```

### CODE REVIEW

File: core/src/io/anuke/mindustry/core/NetClient.java [hunk 1]
```diff
@@ -198,6 +199,16 @@ public static String colorizeName(int id, String name){
...
     }
 
+    @Remote(called = Loc.client, variants = Variant.one)
+    public static void onConnect(String ip, int port){
+        netClient.disconnectQuietly();
+        state.set(State.menu);
+        logic.reset();
+
+        Vars.netClient.beginConnecting();
+        net.connect(ip, port, Actions::hide);
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/core/NetClient.java]: What is `Actions::hide` for? There's nothing to hide here.

Comment (Kieaer) [reply core/src/io/anuke/mindustry/core/NetClient.java]: Can use ``net.connect(ip, port, null);``?

Comment (Anuken) [reply core/src/io/anuke/mindustry/core/NetClient.java]: No, but there's no reason to use an empty UI action.

File: core/src/io/anuke/mindustry/core/NetClient.java [hunk 2]
```diff
@@ -198,6 +199,16 @@ public static String colorizeName(int id, String name){
...
     }
 
+    @Remote(called = Loc.client, variants = Variant.one)
+    public static void onConnect(String ip, int port){
+        netClient.disconnectQuietly();
+        state.set(State.menu);
+        logic.reset();
+
+        Vars.netClient.beginConnecting();
+        net.connect(ip, port, Actions::new);
```
Comment (Anuken) [inline core/src/io/anuke/mindustry/core/NetClient.java]: That's pretty much the same thing. `Actions` is a class for animated UI elements, it has nothing to do with this connections callback. Don't use it.

What you want is an empty callback, which would be `() -> {}`.

**Comment by Kieaer** [comment]
I learned how to call an empty callback by Anuke!
~~knowledge 5 increase~~

**Pull request merged**